metagraph-plugin/metagraph_katana/plugins/metagraph_katana/algorithms.py

```diff
@@ -28,10 +33,12 @@ def kg_bfs_iter(graph: KatanaGraph, source_node: NodeID, depth_limit: int) -> Nu
     :rtype: NumpyVectorType
     '''
     bfs_prop_name = "bfs_prop_start_from_" + str(source_node)
+    depth_limit_internal = 2**30 - 1 if depth_limit == -1 else depth_limit # return all the reachable nodes for the default value of depth_limit (-1)
```

Comment:
2**30 is arbitrary. It would be better to extend the check below to be `pg_bfs_list[i] < depth_limit or depth_limit == -1` or something like that.